src/box/txn_limbo.h

```diff
@@ -169,6 +164,18 @@ struct txn_limbo {
 	 * illegal.
 	 */
 	int64_t confirmed_lsn;
+	/**
+	 * The entry preceding the first unconfirmed synchronous transaction
+	 * or the last entry in the limbo if there are no unconfirmed
+	 * synchronous transactions yet.
+	 */
+	struct txn_limbo_entry *entry_to_confirm;
```

Comment:
The comment seems to be describing the old behaviour. Now this points at the first unconfirmed txn itself, not at the preceding entry. The comment below also says `the transaction is the next one immediately after the entry_to_confirm`, but really `ack_count` is for this entry.

Reply:
Fixed.
```diff
diff --git a/src/box/txn_limbo.h b/src/box/txn_limbo.h
index bbf130bc4..6f434c712 100644
--- a/src/box/txn_limbo.h
+++ b/src/box/txn_limbo.h
@@ -165,15 +165,16 @@ struct txn_limbo {
         */
        int64_t confirmed_lsn;
        /**
-        * The entry preceding the first unconfirmed synchronous transaction
-        * or the last entry in the limbo if there are no unconfirmed
-        * synchronous transactions yet.
+        * The first unconfirmed synchronous transaction in the current term.
+        * Is NULL if there is no such transaction, or if the current instance
+        * does not own limbo.
         */
        struct txn_limbo_entry *entry_to_confirm;
        /**
-        * Number of ACKs of the first unconfirmed synchronous transaction in
-        * limbo (the transaction is the next one immediately after the
-        * entry_to_confirm).
+        * Number of ACKs of the first unconfirmed synchronous transaction
+        * (entry_to_confirm->txn). Contains the actual value only for a
+        * non-NULL entry_to_confirm with a local lsn assigned. Otherwise
+        * it may contain any trash.
         */
        int ack_count;
        /**
```